### ISSUE TITLE
VSCode throws error ⇒ `Hypr client: couldn't create connection to server` even though hyprls is in `$PATH` - Built from source.

### ISSUE THREAD

**Issue by baudneo** [hyprland-community/hyprls#8]
I have built hyprLS from source using Just. I am using `visual-studio-code-bin-1.89.1-2` and `hyprls` is in my PATH.

The full error from vscode when I open `hyprland.conf` is:
`Hypr client: couldn't create connection to server.
Launching server using command hyprls failed. Error: spawn hyprls ENOEN`

If I open the terminal tab and type hyprls, the hyprls server starts. I am unsure of how I can debug this issue as I don't use vscode often, I use jetbrains products but jetbrains has no support for hyprlang.

So, basically, `hyprls` is installed and in my `$PATH`, I can start hyprls by typing `hyprls` into the vscode terminal or my regular terminal.

What am I doing wrong?

Edit:

```
❯ hyprls
2024-05-13T22:27:48.655-0600	DEBUG	hyprls/main.go:25	going to start server
2024-05-13T22:27:48.655-0600	DEBUG	hyprls/main.go:18	starting server
```

**Comment by ShyamGadde** [comment]
I've encountered a similar issue. However, when I navigated to the `.config/hypr` directory and initiated VSCode with the command `code .`, it functioned correctly. The exact cause is unclear, but it seems that Node.js is unable to spawn a new process because it cannot locate `hyprls` when I open the `.config/hypr` directory from inside VSCode.

**Comment by baudneo** [comment]
> I've encountered a similar issue. However, when I navigated to the `.config/hypr` directory and initiated VSCode with the command `code .`, it functioned correctly. The exact cause is unclear, but it seems that Node.js is unable to spawn a new process because it cannot locate `hyprls` when I open the `.config/hypr` directory from inside VSCode.

This worked for me as well. Very strange behaviour, I wonder how this can be worked around for wofi drun? I can't seem to find the .desktop file for vscode.

**Comment by ShyamGadde** [comment]
I don't think you need to do anything with the `.desktop` file. I use `zsh` and I just moved the following from `.zshrc` to `~/.zshenv` and everything works fine now, irrespective of how VSCode was launched.

```sh
#GO
export PATH=$PATH:$(go env GOPATH)/bin
```

**Comment by baudneo** [comment]
Thank you very much for your help, it is greatly appreciated!

**Comment by baudneo** [comment]
Unfortunately, I compiled hyprla from source so using your workaround does not work for me. I have to use the first workaround (cd to `~/.config/hype` and run `code .`).

I will update in future once I figure out a solution.

**Comment by ShyamGadde** [comment]
If you compiled it from source, the most obvious thing I can think of is that `~/.local/bin` is not in your `$PATH`.

So, add something like this in your `~/.zshenv`:
```sh
# Set $PATH if ~/.local/bin exist
if [ -d "$HOME/.local/bin" ]; then
	export PATH=$HOME/.local/bin:$PATH
fi
```

Btw, I'm just assuming you use ZSH, if you are using Bash, it's probably going to be a bit different.

**Comment by baudneo** [comment]
hyperls is already in $PATH using `.zshenv`, still get the same behaviour.

```shell
export PATH=~/.local/bin:"$PATH"
```

Wonder if there is some vscode setting to solve this?

**Comment by baudneo** [comment]
Adding to the VSCode integrated terminal env doesn't seem to do the trick either:

```
    "terminal.integrated.env.linux": {
        "PATH": "/home/baudneo/.local/bin:/home/baudneo/.config/hypr"
      }
```

**Comment by baudneo** [comment]
OK, this is weird, after a reboot it is now working. I am unsure of what fixed it exactly, I am assuming it was the VSCode terminal path from above.
